PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -135,8 +151,14 @@ struct ValidationGenLevel {
           whichHadron = iD;
           RecoDecay::getDaughters(particle, &listDaughters, arrPDGFinal[iD], -1);
           std::size_t arrayPDGsize = arrPDGFinal[iD].size() - std::count(arrPDGFinal[iD].begin(), arrPDGFinal[iD].end(), 0);
+          int origin = 0;
           if (listDaughters.size() == arrayPDGsize) {
-            counter[iD]++;
+            origin = (RecoDecay::getMother(particle, kBottom, true) > -1 ? OriginType::NonPrompt : OriginType::Prompt);
+            if (origin == 4) {
+              counterPrompt[iD]++;
+            } else if (origin == 5) {
+              counterNonPrompt[iD]++;
```

Comment:
You should compare against `OriginType::NonPrompt`, `OriginType::Prompt`. They don't have values 4 and 5.

Reply:
Hi @vkucera thanks for noticing it! Indeed I implemented it a la AliPhysics, all the occurences are now fixed